src/client/activation/jedi/languageServerProxy.ts

```diff
@@ -114,45 +105,46 @@ export class JediLanguageServerProxy implements ILanguageServerProxy {
         interpreter: PythonEnvironment | undefined,
         options: LanguageClientOptions,
     ): Promise<void> {
-        if (!this.languageClient) {
-            this.pythonSettings = this.configurationService.getSettings(resource);
-
-            this.lsVersion =
-                (options.middleware ? (<LanguageClientMiddleware>options.middleware).serverVersion : undefined) ??
-                '0.19.3';
+        if (this.languageClient) {
+            return this.serverReady();
+        }
```

Comment:
At this point can't we just eliminate `this.serverReady()`?

```suggestion
        if (this.languageClient) {
            return this.languageClient.onReady();
        }
```

Reply:
I though about this, the `serverReady` method is protected. I wanted to get rid of that separately, making sure that we don't accidentally break some behavior.